nixos/modules/services/networking/wpa_supplicant.nix

```diff
@@ -155,6 +176,26 @@ in {
         '';
       };
 
+      environmentFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          File consisting of lines of the form <literal>varname=value</literal>
+          to define variables for the wireless configuration.
+
+          See section "EnvironmentFile=" in <literal>systemd.exec(5)</literal>
+          for a syntax reference.
+
+          This option can be used to safely include secrets (PSKs, passwords,
+          etc.) in the <option>networking.wireless.networks</option> option.
+
+          For example, if you define your home wifi password as
```

Comment:
Can you make this into an `example = ''[...]` and give a literal example of a valid file?

Reply:
I thought about this, but how would I show the file content given the option takes a filepath?
I would have to write `pkgs.writeText "secrets" ''...''`, which seems to suggest putting the secrets into the Nix store, again.

Reply:
Ah true, so maybe give as example an example path to such a file, and then say in the description something along the line of "the file `/foo/bar.secrets` should contain contents `@MYPASS@=secret`"?

Reply:
Or should it contain `MYPASS=secret`..? Well, an example would clarify that :)

Reply:
I does say `PSK_HOME=mypassword`. Anyway, I will write a manual chapter with a more detailed explanation, including why this variable dance is needed.

Reply:
> I does say PSK_HOME=mypassword. 

Yes, I'm nitpicking here, what I find unclear is that this is what you need to write in the _file_. (And not e.g. in your config.)